classes/Translate.php

```diff
@@ -194,6 +194,8 @@ public static function getModuleTranslation(
         }
 
         if (empty($iso)) {
+            // using global $_MODULES only when $locale is not set to prevent getting translation in wrong language
```

Comment:
Hi ! I am sorry I dont think this bugfix can be merged. It might solve your issue but it impacts the whole Translate class and I am quite sure there will be unwanted side-effects.

Can you explain the logic behind this PR ? We might find another way that solves your issue without introducing others.

Do you only set `$_MODULES` here because `getIsoByLocale` uses it ?

Reply:
As I've written in the Issue mentioned in this PR, when I use `$this->l()` with locale specified, and there's no string for this locale, I get string in previous language that had translation for this string.
Because `$_MODULES` variable stores it without any mention of language the string is in, I can't know if that's proper translation, or something carried over from previous function call, so only thing that seemed to fix it for me was not using global `$_MODULES` when locale was provided. There might be better way to do this, and the way I did it might introduce bugs elsewhere, but I made this PR anyway, to get the ball rolling.

Reply:
Other thing I thought about is adding another argument, for example `useCache`, that is true by default, but this seemed more like a workaround than real fix.